swarm/storage/memstore_lrucache.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2018 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+// memory storage layer for the package blockhash
+
+package storage
+
+import (
+	"bytes"
+	"fmt"
+	"sync"
+
+	lru "github.com/hashicorp/golang-lru"
+)
+
+type MemStore struct {
+	cache    *lru.Cache
+	requests *lru.Cache
+	mu       sync.Mutex
+	disabled bool
+}
+
+func NewMemStore(_ *LDBStore, cacheCapacity uint, requestsCapacity uint) (m *MemStore) {
```

Comment:
We have a comment next to the default value:
```
	defaultChunkRequestsCacheCapacity = 5000000 // capacity for container holding outgoing requests for chunks. should be set to LevelDB capacity
```

If one is setting the current `StoreParams` they should at least check the default values, and know what these parameters mean.

Then again I agree this is very ugly, and should be fixed very soon.

Reply:
NewMemStore by itself is an exported function, imho this worth an extra line of comment here, you don't necessarily use the default values. But this is not a blocker for sure.

Reply:
Will add a comment. I really feel uneasy about this whole setup, so I am not sure how to document it in a meaningful way. Nevertheless will add a comment so that at least future coders have some intuition about it.